File: Convertor de Temperatura Java/src/dados/Fahrenheit.java
package dados;

public class Fahrenheit extends Termometro
{

	public Fahrenheit(double fahrenheit)
	{
		super(fahrenheit);
	}
	
	public void converte()
	{
		double celsius = (getTemperatura() - 32) * 5 / 9;
		setCelsius(celsius);
		double fahrenheit = getTemperatura() ;
		setFahrenheit(fahrenheit);
		double kelvin = (getTemperatura() - 32) * 5 / 9 + 273;
		setKelvin(kelvin);
	}

}

File: Convertor de Temperatura C++/src/fahrenheit.cpp
#include "fahrenheit.hpp"

void Fahrenheit::converte(float fahrenheit)
{
    cout << "Temperatura em Fahrenheit: " << setiosflags (ios::fixed) << setprecision (1) << fahrenheit << " ºF" << endl;
    cout << "Temperatura em Celsius: " << setiosflags (ios::fixed) << setprecision (1) << (fahrenheit - 32) * 5 / 9 << " ºC" << endl;
    cout << "Temperatura em Kelvin: " << setiosflags (ios::fixed) << setprecision (1) << (fahrenheit - 32) * 5 / 9 + 273 << " K" << endl;
}
